PhysicsTools/PatAlgos/plugins/PATSecondaryVertexSlimmer.cc

```diff
@@ -32,6 +37,7 @@ namespace pat {
 pat::PATSecondaryVertexSlimmer::PATSecondaryVertexSlimmer(const edm::ParameterSet& iConfig) :
   src_(consumes<reco::VertexCompositePtrCandidateCollection>(iConfig.getParameter<edm::InputTag>("src"))),
   srcLegacy_(mayConsume<std::vector<reco::Vertex> >(iConfig.getParameter<edm::InputTag>("src"))),
+  srcV0s_(mayConsume<reco::VertexCompositeCandidateCollection>(iConfig.getParameter<edm::InputTag>("src"))),
```

Comment:
this runtime input type detection is getting only worse.
Please change the module configuration to pass the type to be used at configuration step.

@Dr15Jones is proliferation of this mayConsume to different types OK for the framework, or do we get some unwanted cost?